design_drafts/components_architecture_and_urdf_examples.md

```diff
@@ -0,0 +1,424 @@
+# ROS2 Control Components Architecture and URDF-Description Examples
+
+## Architecture
+
+![ros2_control Components Achitecture][ros2_control_arch_resource_manager]
+
+[ros2_control_arch_resource_manager]: images/components_architecture.png "ROS2 Control - Components Architecture"
+
+## URDF Examples
+
+ros2_control implementation examples are presented for the following robot/robot-cell architectures:
+
+1. Industrial Robots with only one interface
+2. Industrial Robots with multiple interfaces (can not be written at the same time)
+3. Robots with multiple interfaces used at the same time
+4. Industrial Robots with a sensor integrated into the robot's control box
+5. Industrial Robots with a sensor connected to ROS computer
+6. Modular Robots with separate communication to each actuator
+7. Modular Robots with actuators not providing states and additional sensors (simple Transmissions)
+8. Modular Robots with separate communication to each "actuator" with multi joints (Transmission Example) - (system component is used)
+9. Sensor only
+10. Actuator Only
+
+
+Note:
+  * Everything within the `<classType>` tag is implemented as a plugin.
+  * The examples below have some `<param>` tags defined for each plugin which are primarily for demonstration, not part of a pre-defined xml schema. Components can define their own parameters.
```

Comment:
I reword this. Is it better now?

> But we have to maybe clarify how these parameters are being used. 

This would be done in the example implementation.